src/core/expr/fexpr_cumsum.cc

```diff
@@ -96,7 +120,7 @@ class FExpr_cumsum : public FExpr_Func {
     template <typename T>
     Column make(Column&& cumsum, SType stype0) const {
       cumsum.cast_inplace(stype0);
-      return Column(new Latent_ColumnImpl(new Column_cumsum<T>(std::move(cumsum))));
+      return Column(new Column_cumsum<T>(std::move(cumsum)));
```

Comment:
It is the `Latent_ColumnImpl` that will call `.materialize()` for the `Column_cumsum` column on the first data access. So you should wrap `Column_cumsum` with the `Latent_ColumnImpl` as it was before.

Reply:
Btw, the class should be named `Cumsum_ColumnImpl`, because it is inherited from `Virtual_ColumnImpl`.